### PR TITLE
Layout Algorithm for Rectangles

### PR DESCRIPTION
**Pull request by ArquintL**
This PR adds `arquint`, an algorithm that treats nodes as rectangles each having a property `heightRatio` that defines the height relative to other nodes.

In addition to mapping nodes to layers, they also get mapped to columns. This helps in assigning x0, x1, y0, and y1 coordinates to each node. 4 different variants of column assignment accessors are included in this PR.

All details can be found in the extended README (including some nice screenshots)

### CODE REVIEW

**Comment by erikbrinkman** [comment]
Hi @ArquintL, this looks really cool! I have been busy lately, so I haven't had time to look over it yet, but I know how frustrating it is to submit an awesome PR only to have the maintainers drag their feet. I'll try to get to this soon, but if I don't, feel free to hound me.

**Comment by ArquintL** [comment]
@erikbrinkman I intended to create some snipped on observablehq to play around with the parameters (like your example). However, I gave up importing an unreleased npm module ^^

In order to create the screenshots I replicated the example in a separate repo, in case you wanna play around: https://github.com/ArquintL/d3-dag-example

File: README.md [hunk 1]
```diff
@@ -646,3 +647,147 @@ Lays out the specified DAG, assigning the following properties:
 
 If *size* is specified, sets this zherebko layout's size to the specified two-element array of numbers [*width*, *height*] and returns this zherebko layout.
 If *size* is not specified, returns the current layout size, which defaults to [1, 1].
+
+
+### Arquint
+
+This treats nodes not as points (i.e. producing x & y coordinates) but as rectangles.
+Each node has a property *heightRatio* specifying its height in comparison to other nodes.
+The algorithm is based off a PR by L. Arquint.
```
Comment (erikbrinkman) [inline README.md]: I would rephrase this a little. Saying it's based off a PR is a little inaccurate because it was submitted as a PR in full. First, I would link to or reference either a publication or further material on the algorithm potentially a link to your personal website. If you want to emphasize the fact that you also wrote the corresponding code, then I would say something like, "This implementation was contributed by the author."

**Comment by ArquintL** [comment]
@erikbrinkman 
I've changed the wording in README to better reflect that I'm the author.
Furthermore, I tried to spend a few more words about the algorithms to assign column indices. Unfortunately, I do not have any publications about these because my current focus is somewhere completely different (visualizing logs from a symbolic execution engine).

A few comments about your general suggestions:
1. I've added now a few tests that verify that column assignment as well as translation from column indices to coordinates works as expected
2. `coordRect` is quite different from the existing `coord` functions. Therefore, I've renamed `coordRect` to `column2CoordRect` to emphasize that it actually only translates from column indices to coordinates (for rectangles)
3. I need the typings to use the library from my typescript project. However, I've removed it from the PR and will use it separately. Furthermore, best practice for typings is to place them [here](https://github.com/DefinitelyTyped/DefinitelyTyped). Because my current typings are not complete yet (for the entire d3-dag library), I haven't published them there yet. But in case there is interest to further develop them, I'm happy to assist
4. done
5. I tried my best to get as close as possible to the existing images, but I also think that having an interactive example on observable would be most helpful
